session/general_log.go

```diff
@@ -0,0 +1,196 @@
+// Copyright 2021 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package session
+
+import (
+	"strings"
+	"sync"
+	"time"
+	"unsafe"
+
+	"github.com/natefinch/lumberjack"
+	"github.com/pingcap/parser/ast"
+	"github.com/pingcap/tidb/br/pkg/lightning/log"
+	"github.com/pingcap/tidb/metrics"
+	"github.com/pingcap/tidb/sessionctx/stmtctx"
+	"github.com/pingcap/tidb/sessionctx/variable"
+	"go.uber.org/zap"
+	"go.uber.org/zap/zapcore"
+)
+
+var (
+	// GeneralLogLogger is used to log general log
+	GeneralLogLogger       = log.L()
+	globalGeneralLogger    *generalLogger
```

Comment:
It's generally not a best practice to use something global. Could you try to make it non-global? For example, maybe it can be a Domain level instance.